GVRf/Framework/framework/src/main/java/org/gearvrf/x3d/AnimationInteractivityManager.java

```diff
@@ -628,17 +663,122 @@ public void onSensorEvent(SensorEvent event) {
-                                        if (interactiveObject.getEventUtility() == interactiveObjectFinal.getEventUtility()) {
-                                            GVRSceneObject gvrSceneObject = root
-                                                    .getSceneObjectByName(interactiveObject.getDefinedItem().getName());
-                                            GVRComponent gvrComponent = gvrSceneObject.getComponent(GVRLightBase.getComponentType());
-                                            gvrComponent.setEnable(eventUtility.getToggle());
+                                    if ( eventUtility != null ) {
+                                        eventUtility.setToggle(!eventUtility.getToggle());
+                                        for (InteractiveObject interactiveObject : interactiveObjectsFinal) {
+                                            if (interactiveObject.getEventUtility() == interactiveObjectFinal.getEventUtility()) {
+                                                GVRSceneObject gvrSceneObject = root
+                                                        .getSceneObjectByName(interactiveObject.getDefinedItem().getName());
+                                                GVRComponent gvrComponent = gvrSceneObject.getComponent(GVRLightBase.getComponentType());
+                                                gvrComponent.setEnable(eventUtility.getToggle());
+                                            }
+                                        }
+                                    }
+                                    else {
+                                        // this is an fbx or other non-X3D file for animation
+                                        if ( !activeFBXanimation ) {
+                                            InlineObject inlineObject = interactiveObjectFinal.getInlineObject();
+                                            GVRSceneObject inlineSceneObject = inlineObject.getInlineGVRSceneObject();
+                                            GVRAnimator animator = (GVRAnimator) inlineSceneObject.getComponent(GVRAnimator.getComponentType());
+                                            if ( animator != null ) {
+                                                List<GVRAnimation> gvrAnimations = animator.getAnimationList();
+                                                GVRAnimation gvrAnimation = null;
+
+                                                if  ( (inlineObject.getBeginKeyFrame() == InlineObject.UNSET_KEY_FRAME) || (inlineObject.getEndKeyFrame() == InlineObject.UNSET_KEY_FRAME)) {
+                                                    if ( (inlineObject.getBeginKeyFrame() == InlineObject.UNSET_KEY_FRAME) && (inlineObject.getEndKeyFrame() == InlineObject.UNSET_KEY_FRAME)) {
+                                                        // use the original FBX animation.
+                                                        gvrAnimation = gvrAnimations.get(0);
```

Comment:
What if there are no animations in the list? What are there guarantees there will be at least one?